From: Jiri Bohac <jbohac@suse.cz>
Subject: net: add BUG_ON if kernel advertises msg_namelen > sizeof(struct
Git-commit: 68c6beb373955da0886d8f4f5995b3922ceda4be
Patch-mainline: v3.13
References: bnc#853040

In that case it is probable that kernel code overwrote part of the
stack. So we should bail out loudly here.

This is a BUG_ON upstream (68c6beb373955da0886d8f4f5995b3922ceda4be).
Be nicer by making this a WARN_ON in a maintenance update. 


diff --git a/net/socket.c b/net/socket.c
index d01b915..a51241a 100644
--- a/net/socket.c
+++ b/net/socket.c
@@ -215,6 +215,8 @@ static int move_addr_to_user(struct sockaddr *kaddr, int klen,
 	int err;
 	int len;
 
+	WARN_ON(klen > sizeof(struct sockaddr_storage));
+
 	err = get_user(len, ulen);
 	if (err)
 		return err;
